docs/extensibility/annotations.md

```diff
@@ -0,0 +1,65 @@
+# Annotations
+
+Annotations are a way to highlight a specific piece in a Gutenberg post. Examples of this include commenting on a piece of text and spellchecking. Both can use the annotations API to mark a piece of text.
+
+## API
+
+To see the API for yourself the easiest way is to have a block that is at least 200 characters long without formatting and putting the following in the console:
+
+```js
+wp.data.dispatch( 'core/editor' ).addAnnotation( {
+	source: "my-annotations-plugin",
+	block: wp.data.select( 'core/editor' ).getBlockOrder()[0],
+	startXPath: "text()[1]",
```

Comment:
Do you think it would be possible to group all XPath related properties under one property? I don't have anything against using XPath, but maybe we could make it flexible enough so we could easily swap it with something else when it fits better.

Is there any particular relation between those 4 properties:
- `startXPath`
- `startOffset`
- `endXPath`
- `endOffset`

It particularly confusing in the second example where `startOffset` is greater than `endOffset`.